web/modules/custom/pulses_exchange_rate/src/Form/ExchangeApiKey.php

```diff
@@ -0,0 +1,181 @@
+  /**
+   * Custom config name.
+   *
+   * @var string
+   */
+  const SETTINGS = 'pulses_exchange_rate.settings';
+
+  /**
+   * API URl. Prepared to concat with  parameter called api key.
+   *
+   * @var string
+   */
+  const API_URL = 'https://api.currencyapi.com/v3/latest?apikey=';
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getFormId() {
+    return 'pulses_exchange_rates';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      static::SETTINGS,
+    ];
+  }
+
+  /**
+   * Form for select name of currency.
+   *
+   * Currency Data Provider provide possibility to check and get data from api
+   * for form select elements in 'currency' fieldset wrapper.
+   *
+   * {@inheritDoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config(static::SETTINGS);
+    $api_key = $form_state->getValue('api_key');
+    $valid_key = preg_match('/^.{40}$/', $api_key);
+    $valid_response = FALSE;
+    if (!$valid_key) {
+      $api_key = $config->get('key');
+    }
+    else {
+      $valid_response = $this->render->getResponse(static::API_URL, $api_key, FALSE);
+    }
+    $form['api_key'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Please, enter your Api key:'),
+      '#default_value' => $config->get('key'),
+      '#maxlength' => 40,
+      '#description' => $this->t('Press enter for continue filling form'),
+      '#ajax' => [
+        'callback' => '::validateApiKey',
+        'disable-refocus' => TRUE,
+        'keypress' => TRUE,
+        'event' => 'change',
+        'wrapper' => 'currency-wrapper',
+        'progress' => FALSE,
+      ],
+    ];
+    $form['currency'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'id' => 'currency-wrapper',
+      ],
+    ];
+    if (!empty($api_key) && $valid_response) {
+      $form['currency']['first'] = [
+        '#type' => 'select',
+        '#title' => $this->t('Select needed currency:'),
+        '#options' => $this->render->data,
+        '#default_value' => $config->get('currency.first'),
+      ];
+      $form['currency']['second'] = [
+        '#type' => 'select',
+        '#title' => $this->t('Select needed currency:'),
+        '#options' => $this->render->data,
+        '#default_value' => $config->get('currency.second'),
+      ];
+      $form['currency']['third'] = [
+        '#type' => 'select',
+        '#title' => $this->t('Select needed currency:'),
+        '#options' => $this->render->data,
+        '#default_value' => $config->get('currency.third'),
+      ];
+      $form['currency']['actions']['#type'] = 'actions';
+      $form['currency']['actions']['submit'] = [
+        '#type' => 'submit',
+        '#value' => $this->t('Save configuration'),
+        '#button_type' => 'primary',
+      ];
+
+      // By default, render the form using system-config-form.html.twig.
+      $form['#theme'] = 'system_config_form';
+    }
```

Comment:
This code should be always executed.
```suggestion
    }
    // By default, render the form using system-config-form.html.twig.
    $form['#theme'] = 'system_config_form';
```